web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -174,58 +173,26 @@ const Dashboard = (props: Props) => {
   const { app, isBundleUploading, isVeleroInstalled } = props;
   const airgapUploader = useRef<AirgapUploader | null>(null);
 
-  const fetchAppDownstream = async () => {
-    if (!app) {
-      return;
-    }
-
-    try {
-      const res = await fetch(`${process.env.API_ENDPOINT}/app/${app.slug}`, {
-        headers: {
-          Authorization: Utilities.getToken(),
-          "Content-Type": "application/json",
-        },
-        method: "GET",
-      });
-      if (res.ok && res.status == 200) {
-        const appResponse = await res.json();
-        if (!isAwaitingResults(appResponse.downstream.pendingVersions)) {
-          state.fetchAppDownstreamJob.stop();
-        }
-        setState({
-          downstream: appResponse.downstream,
-        });
-        // wait a couple of seconds to avoid any race condiditons with the update checker then refetch the app to ensure we have the latest everything
-        // this is hacky and I hate it but it's just building up more evidence in my case for having the FE be able to listen to BE envents
-        // if that was in place we would have no need for this becuase the latest version would just be pushed down.
-        setTimeout(() => {
-          props.refreshAppData();
-        }, 2000);
-      } else {
-        setState({
-          loadingApp: false,
-          gettingAppErrMsg: `Unexpected status code: ${res.status}`,
-          displayErrorModal: true,
-        });
-      }
-    } catch (err) {
-      console.log(err);
-      const errorMessage =
-        err instanceof Error
-          ? err.message
-          : "Something went wrong, please try again.";
-      setState({
-        loadingApp: false,
-        gettingAppErrMsg: errorMessage,
-        displayErrorModal: true,
-      });
-    }
+  const onAppDownstreamSuccess = (data: Downstream) => {
+    setState({ downstream: data });
+    setTimeout(() => {
+      props.refreshAppData();
+    }, 2000);
```

Comment:
minor nit. its best to `clearTimeout` when the component unmounts. I don't think it's a big deal here unless the handlers are going to blow up if they receive data after the component is unmounted https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout